### ISSUE TITLE
Optimizing only one variable

### ISSUE THREAD

**Issue by emcastillo** [JasperSnoek/spearmint#51]
Hello,

I am currently using this version of spearmint as I have spearmint-lite integrated in our own framework to tune neural networks hyper-parameters. We only require a tool for proposing us the points to evaluate and we control all the concurrency in our framework. 

I have encountered an issue when I tried to optimize only one variable with 1 dimension.
The learning rate of our neural network.

After some debugging I identified an issue in the chooser/GPEIOptChooser.py file line 511

https://github.com/JasperSnoek/spearmint/blob/master/spearmint/spearmint/chooser/GPEIOptChooser.py

grad_cross = np.squeeze(cand_cross_grad)

When dealing with only 1D variable, the cand_cross_grad always have a shape of (N,1,1) that becomes (N,) when squeezed. Further operations using this values leads to matrices with erroneous sizes that the scikit bfgs optimization can not handle.

I have done a quick fix to leave the cand_cross as (N,1) only in that situation:
```
if pend.shape[1]==1:
    grad_cross = np.squeeze(cand_cross_grad,axis=(2,))
else:
    grad_cross = np.squeeze(cand_cross_grad)
```
We had not extensively tested it yet but we just wanted to share it in case that anyone else finds this issue

**Comment by JasperSnoek** [comment]
Thanks very much @emcastillo!  The single dimensional case is uncommon, but it's certainly important that it works.  Would you like to include this as a pull request?

**Comment by emcastillo** [comment]
Done! Thanks a lot for this amazing tool, we love it.
